src/uwtools/rocoto.py

```diff
@@ -182,13 +182,56 @@ def _add_task_dependency(self, e: Element, config: dict) -> None:
         :param config: Configuration data for this element.
         """
         e = SubElement(e, STR.dependency)
-        for key, block in config.items():
-            tag, _ = self._tag_name(key)
-            if tag == STR.taskdep:
-                self._set_attrs(SubElement(e, STR.taskdep), block)
+        for tag, block in config.items():
+            tag, _ = self._tag_name(tag)
+            if tag in [STR.taskdep, STR.datadep, STR.timedep]:
+                self._add_task_dependency_operand(e, tag=tag, block=block)
+            elif tag in [STR.and_, STR.nand, STR.nor, STR.not_, STR.or_, STR.xor]:
+                self._add_task_dependency_operator(e, config={tag: block})
+            elif tag in [STR.streq, STR.strneq]:
+                self._add_task_dependency_strequality(e, tag=tag, block=block)
             else:
                 raise UWConfigError("Unhandled dependency type %s" % tag)
 
+    def _add_task_dependency_operand(self, e: Element, tag: str, block: dict) -> None:
+        """
+        :param e: The parent element to add the new element to.
+        :param tag: Configuration new element to add.
+        :param block: Configuration data for the tag.
+        """
+        self._set_attrs(SubElement(e, tag), block)
+
+    def _add_task_dependency_operator(self, e: Element, config: dict) -> None:
+        """
+        Add an operator  element to the <dependency>.
```

Comment:
nitpick: double space between `operator` and `element`

Reply:
Should be fixed with the latest push.